ninjato-core/src/main/kotlin/com/agoda/ninjato/coroutine/CancellableApi.kt

```diff
@@ -0,0 +1,62 @@
+package com.agoda.ninjato.coroutine
+
+import com.agoda.ninjato.Api
+import com.agoda.ninjato.http.HttpClient
+import com.agoda.ninjato.http.Method
+import com.agoda.ninjato.http.Request
+
+abstract class CancellableApi(
+        client: HttpClient,
+        config: Api.() -> Unit = {}
+): Api(client, config) {
+
+    /**
+     * Executes the GET request with the provided configuration.
+     *
+     * @param configurator the configuration of a [Request]
+     */
+    suspend inline fun <reified T> Get(configurator: Request.Configurator.() -> Unit): T = prepare(
```

Comment:
Since I can't override the `get` method, I named it as `Get` to avoid compilation issue. Maybe we can think of a different name.